employees/views.py

```diff
@@ -45,25 +45,26 @@ def perform_create(self, serializer: ReportSerializer) -> None:
         serializer.save(author=self.request.user)
 
 
-def query_as_dict(query_set: QuerySet) -> Dict[str, Any]:
-    dictionary = {}  # type: Dict[str, Any]
-    for record in query_set:
-        key = record.date
-        dictionary.setdefault(key, [])
-        dictionary[key].append(record)
-    return dictionary
+def daily_hours_sum(queryset: QuerySet) -> dict:
```

Comment:
This should be `ReportManager` method. We don't have one yet so please create one for Reports. Here is more info on it https://docs.djangoproject.com/en/2.2/topics/db/managers/#custom-managers. It will be a nice exercise too.

Reply:
I have a couple of questions, I'd like to get out of the way first:
1. As far as I understand, a manager should return a QuerySet object. This means I shouldn't create a function returning dictionary. Shall I then cast each call in the view with `dict()`? Wouldn't this be less clean  and convenient?
2. QuerySets used in this calculation may vary from view to view. Should I define each set in the manager as methods? Could I use parameters in manager's methods?

This solution seems somewhat restrictive, should I follow conventions. I get why it's worth to move them to a manager, but at the moment I'm not sure how to properly handle this one.

Reply:
> As far as I understand, a manager should return a QuerySet object. This means I shouldn't create a function returning dictionary. Shall I then cast each call in the view with dict()? Wouldn't this be less clean and convenient?

You don't have to return `QuerySet` from `Manager`, it can be anything (just as you did). The wider concept of that is that there is `models.QuerySet` and `models.Manager`. Their responsibilities are well described here https://docs.djangoproject.com/en/2.2/topics/db/managers/. But also as you can find in this doc, it says

> In lieu of the above approach which requires duplicating methods on both the QuerySet and the Manager, QuerySet.as_manager() can be used to create an instance of Manager with a copy of a custom QuerySet’s methods:

This is what we do here for simplicity. I have also find this to be common practice on most of the projects I have seen (I mean I have never seen in production code methods being split between `models.QuerySet` and `models.Manager`.

> QuerySets used in this calculation may vary from view to view. Should I define each set in the manager as methods? Could I use parameters in manager's methods?

Yes, you can pass parameters. I think the answer for first part of the question is "it depends on a case". Sometimes its better to write separate method, sometimes its better to parametrize. You need to follow your common sense, experience and project specifics.

Reply:
Thanks for detailed response. Previously I might've written few counter-intuitive functions, BECAUSE I took to much liberty. So I wanted to be sure my implementation wouldn't go against common practices.

But then I saw your code which you sent me and my doubts evaporated. 😃